packages/zoe/contracts/simpleExchange.js

```diff
@@ -22,63 +15,53 @@ import {
 // support partial fills of orders.
 
 export const makeContract = harden((zoe, terms) => {
-  const sellOfferHandles = [];
-  const buyOfferHandles = [];
+  const sellInviteHandles = [];
+  const buyInviteHandles = [];
+  const { assays } = terms;
+  const {
+    rejectOffer,
+    hasValidPayoutRules,
+    swap,
+    areAssetsEqualAtIndex,
+  } = makeHelpers(zoe, assays);
 
-  const simpleExchange = harden({
-    addOrder: async escrowReceipt => {
-      const {
-        offerHandle,
-        offerRules: { payoutRules },
-      } = await zoe.burnEscrowReceipt(escrowReceipt);
-
-      // Is it a valid sell offer?
-      const sellOfferKinds = ['offerAtMost', 'wantAtLeast'];
-      if (hasValidPayoutRules(sellOfferKinds, terms.assays, payoutRules)) {
-        // Save the valid offer
-        sellOfferHandles.push(offerHandle);
-
-        // Try to match
-        const {
-          offerHandles: activeBuyHandles,
-          payoutRulesArray: activeBuyPayoutRules,
-        } = getActivePayoutRules(zoe, buyOfferHandles);
-        for (let i = 0; i < activeBuyHandles.length; i += 1) {
-          if (isMatchingLimitOrder(zoe, payoutRules, activeBuyPayoutRules[i])) {
-            return reallocate(zoe, offerHandle, activeBuyHandles[i]);
+  const makeInvite = () => {
+    const seat = harden({
+      addOrder: () => {
+        // Is it a valid sell offer?
+        if (hasValidPayoutRules(['offerAtMost', 'wantAtLeast'], inviteHandle)) {
+          // Save the valid offer and try to match
+          sellInviteHandles.push(inviteHandle);
+          for (const buyHandle of zoe.getOfferStatuses(buyInviteHandles)
+            .active) {
+            if (areAssetsEqualAtIndex(0, inviteHandle, buyHandle)) {
+              swap(inviteHandle, buyHandle);
+            }
           }
+          return defaultAcceptanceMsg;
```

Comment:
This confused me at first, because there are two possible meanings of "The offer has been accepted". You mean here that the contract has accepted the offer into the pool, i.e., that the contract has not rejected the offer. However, when I saw `swap` return the same message, I interpreted it to mean that someone took what I was offering in exchange for something I needed, i.e., that the proposed deal had been accepted.

Not sure what to do about that, but would be good to find a less misunderstandable phrase. They can still both use the same phrase.

Reply:
Ok, that's good to know. Part of the background on "userFlow" is that it was meant to consolidate the patterns I was seeing - of either rejecting or accepting an offer. So that's where the defaultAcceptanceMsg came from.

Reply:
Let's revisit this after we finish up this invites sprint. In the next phases of trying to allow for any number of assays, I'm not sure what will still be around and what won't.